### PR TITLE
Add log with OS app string representation to Analysis._populate

### PR DESCRIPTION
**Pull request by rgildein**
With this, we can collect info about application during Analysis._populate, like this the log will contain the app string representation before planning.

log message example:
```bash
found OpenStack application:
app-name:
  model_name: <model-name>
  ...
```

### CODE REVIEW

File: cou/steps/analyze.py [hunk 1]
```diff
@@ -121,10 +121,12 @@ async def _populate(cls, model: juju_utils.Model) -> list[OpenStackApplication]:
         :rtype: List[OpenStackApplication]
         """
         juju_applications = await model.get_applications()
-        apps = {AppFactory.create(app) for app in juju_applications.values()}
+        apps = set()
+        for name, app in juju_applications.items():
+            if os_app := AppFactory.create(app):
+                apps.add(os_app)
+                logger.info("found OpenStack application:\n%s", os_app)
```
Comment (aieri) [inline cou/steps/analyze.py:128]:
```suggestion
                logger.info("Found OpenStack application:\n%s", os_app)
```

File: cou/steps/analyze.py [hunk 2]
```diff
@@ -121,24 +121,26 @@ async def _populate(cls, model: juju_utils.Model) -> list[OpenStackApplication]:
...
-        apps = {AppFactory.create(app) for app in juju_applications.values()}
+        apps = set()
+        for name, app in juju_applications.items():
+            if os_app := AppFactory.create(app):
+                apps.add(os_app)
+                logger.info("Found %s application:\n%s", name, os_app)
 
-        # remove non-supported charms that return None on AppFactory.create
-        apps.discard(None)
         # mypy complains that apps can have None, but we already removed.
```
Comment (agileshaw) [inline cou/steps/analyze.py:130]: we need to remove this comment too

File: cou/steps/analyze.py [hunk 3]
```diff
@@ -121,24 +121,26 @@ async def _populate(cls, model: juju_utils.Model) -> list[OpenStackApplication]:
...
-            app for app in apps if app.charm in UPGRADE_ORDER  # type: ignore
+            app for app in apps if app.charm in UPGRADE_ORDER
         }
         other_o7k_apps = apps - apps_to_upgrade_in_order
         sorted_apps_to_upgrade_in_order = sorted(
             apps_to_upgrade_in_order,
-            key=lambda app: UPGRADE_ORDER.index(app.charm),  # type: ignore
+            key=lambda app: UPGRADE_ORDER.index(app.charm),
         )
         # order by charm name to have a predictable upgrade sequence of others o7k charms.
```
Comment (agileshaw) [inline cou/steps/analyze.py:139]:
```suggestion
        # order by charm name to have a predictable upgrade sequence of other o7k charms.
```

**Review by agileshaw — CHANGES_REQUESTED** [review]
Two minor things

**Comment by aieri** [comment]
superseded by #371

**Pull request closed without merging**